packages/guides/src/Compiler/Passes/GlobalMenuPass.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+declare(strict_types=1);
+
+namespace phpDocumentor\Guides\Compiler\Passes;
+
+use phpDocumentor\Guides\Compiler\CompilerContext;
+use phpDocumentor\Guides\Compiler\CompilerPass;
+use phpDocumentor\Guides\Nodes\DocumentNode;
+use phpDocumentor\Guides\Nodes\Menu\NavMenuNode;
+use Throwable;
+
+class GlobalMenuPass implements CompilerPass
+{
+    public function getPriority(): int
+    {
+        return 20; // must be run very late
+    }
+
+    /**
+     * @param DocumentNode[] $documents
+     *
+     * @return DocumentNode[]
+     */
+    public function run(array $documents, CompilerContext $compilerContext): array
+    {
+        $projectNode = $compilerContext->getProjectNode();
+        try {
+            $rootDocumentEntry = $projectNode->getRootDocumentEntry();
+        } catch (Throwable) {
+            // Todo: Functional tests have not root document entry
+            return [];
+        }
```

Comment:
To be addressed in a follow up, I think it makes sense to remove the exception that is been thrown by `getRootDocumentEntry` if this is a common pattern. So we use exceptions just for exception.